### PR TITLE
refactor: Remove the use of sdk LegacyMsg

### PR DESCRIPTION
**Pull request by JulianToledano**
Closes:
#98 

This PR replaces the use of `sdk.LegacyMsg.GetSigners()` in favor of `codec.GetMsgV1Signers`.

### CODE REVIEW

File: converter.go [hunk 1]
```diff
@@ -163,12 +161,11 @@ func (c converter) UnsignedTx(ops []*rosettatypes.Operation) (tx authsigning.Tx,
 			return nil, crgerrs.WrapError(crgerrs.ErrCodec, err.Error())
 		}
 
-		legacyMsg, ok := msg.(sdk.LegacyMsg)
-		if !ok {
-			return nil, crgerrs.WrapError(crgerrs.ErrCodec, "Failed asserting LegacyMsg type")
+		signers, _, err := c.cdc.GetMsgV1Signers(msg)
+		if err != nil {
+			return nil, err
```
[the remark this answers is not in the document]
Comment (raynaudoe) [reply converter.go:166]: The rabbit is right on this one, the error should be wrapped for clarity sake

File: converter.go [hunk 2]
```diff
@@ -239,24 +236,23 @@
...
 	if err != nil {
 		return nil, crgerrs.WrapError(crgerrs.ErrConverter, fmt.Sprintf("while getting meta from message %s", err.Error()))
 	}
 
-	legacyMsg, ok := msg.(sdk.LegacyMsg)
-	if !ok {
-		return nil, crgerrs.WrapError(crgerrs.ErrCodec, "Failed asserting LegacyMsg type")
+	signers, _, err := c.cdc.GetMsgV1Signers(msg)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (raynaudoe) [reply converter.go:246]: Same here

File: converter.go [hunk 3]
```diff
@@ -239,24 +236,23 @@ func (c converter) Meta(msg sdk.Msg) (meta map[string]interface{}, err error) {
...
 	}
 
-	ops := make([]*rosettatypes.Operation, len(legacyMsg.GetSigners()))
-	for i, signer := range legacyMsg.GetSigners() {
+	ops := make([]*rosettatypes.Operation, len(signers))
+	for i, signer := range signers {
+		addr, _ := c.cdc.InterfaceRegistry().SigningContext().AddressCodec().BytesToString(signer)
```
Comment (raynaudoe) [inline converter.go:251]: is there a risk of encountering a nil pointer dereference at any point in this sequence?

Comment (JulianToledano) [reply converter.go]: It should not, the codec already uses that internally, although we could make a check when initialising rosetta to make sure SigningContext is correctly set.

**Review by lucaslopezf — APPROVED** [review]

**Pull request merged**